v0/destinations/sendgrid/util.js

```diff
@@ -361,12 +413,185 @@ const generatePayloadFromConfig = (payload, Config) => {
   return updatedPayload;
 };
 
+/**
+ * Validation for an email
+ * @param {*} message
+ * @returns
+ */
+const validateIdentifyPayload = message => {
+  const email = getFieldValueFromMessage(message, "email");
+  if (!email) {
+    throw new ErrorBuilder()
+      .setMessage("[SendGrid] :: Parameter mail is required")
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+      })
+      .build();
+  }
+};
+
+/**
+ * Returns address in string format
+ * @param {*} address
+ * @returns
+ */
+const flattenAddress = address => {
+  const addressType = typeof address;
+  let companyAddress = "";
+  if (addressType === "object") {
+    const keys = Object.keys(address);
+    keys.forEach(key => {
+      companyAddress += `${address[key]} `;
+    });
+  } else {
+    companyAddress = address;
+  }
+  return companyAddress.trim();
+};
+
+/**
+ * Returns contactListIds in string format
+ * contactListIds : it's a string created from an array
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const getContactListIds = (message, destination) => {
+  const { Config } = destination;
+  const contactListIds = [];
+  const listIds = getDestinationExternalID(message, "listIds");
+
+  if (Config.listId) {
+    contactListIds.push(Config.listId);
+  }
+
+  if (listIds && typeof listIds === "string") {
+    contactListIds.push(listIds);
+  }
+
+  if (listIds && typeof listIds === "object" && listIds.length > 0) {
```

Comment:
We should check the typeof listIds as an array, right?

Reply:
no, in js if listIds is an array then it's type will be object because in js array is object

Reply:
Got it, but we can use Array.isArray().